src/PrestaShopBundle/Entity/Repository/LogRepository.php

```diff
@@ -27,35 +27,24 @@
 namespace PrestaShopBundle\Entity\Repository;
 
 use Doctrine\DBAL\Connection;
-use Doctrine\DBAL\Query\QueryBuilder;
-use PrestaShop\PrestaShop\Core\Grid\Query\DoctrineQueryBuilderInterface;
-use PrestaShop\PrestaShop\Core\Grid\Query\DoctrineSearchCriteriaApplicatorInterface;
-use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
 use PrestaShop\PrestaShop\Core\Repository\RepositoryInterface;
 
 /**
  * Retrieve Logs data from database.
  */
-class LogRepository implements RepositoryInterface, DoctrineQueryBuilderInterface
+class LogRepository implements RepositoryInterface
 {
     private $connection;
     private $databasePrefix;
     private $logTable;
 
-    /**
-     * @var DoctrineSearchCriteriaApplicatorInterface
-     */
-    private $searchCriteriaApplicator;
-
     public function __construct(
         Connection $connection,
-        $databasePrefix,
-        DoctrineSearchCriteriaApplicatorInterface $searchCriteriaApplicator
+        $databasePrefix
```

Comment:
BC Break

Reply:
Yes ! Delegate this responsibility to a queryBuilder instaed of a repository

Reply:
Could you document it in Pull request ?

Check my PR (#20118) for documentation.

Reply:
Done !